### DISCUSSION TITLE
Storing Image Names In Metadata?

### DISCUSSION THREAD

**Discussion by MikeTJensen** [lllyasviel/ControlNet#399] [General]
Hello!

First I want to say, ControlNet is so freaking good! Absolutely love it. 

I wanted to make one minor change locally so that my Controlnet would store the Image Names of each enabled ControlNet inside the generated image's metadata, so it's easier to reproduce results.

Right now I'm needing to do a lot of organization on my end to make sure I can reproduce a result, since the info generated in the metadata isn't enough. 

Could someone point me to where something like this could be done? I looked in the code but couldn't quite figure it out. 

Thank you!
Mike

**Comment by geroldmeisinger** [comment]
most Stable Diffusion UIs already store metainformation (prompt, workflow etc.) in their image generations. you can store anything you want in there, even images-in-images-in-images... but this is something which has to be implemented by the UI. you should open a feature request in the corresponding UI project. but please search first if an extension like this exists already, it sounds like it might well be.

also see similar discussion here: https://github.com/lllyasviel/ControlNet/discussions/505

**Comment by OrganicBeej** [comment]
> Hello!
> 
> First I want to say, ControlNet is so freaking good! Absolutely love it.
> 
> I wanted to make one minor change locally so that my Controlnet would store the Image Names of each enabled ControlNet inside the generated image's metadata, so it's easier to reproduce results.
> 
> Right now I'm needing to do a lot of organization on my end to make sure I can reproduce a result, since the info generated in the metadata isn't enough.
> 
> Could someone point me to where something like this could be done? I looked in the code but couldn't quite figure it out.
> 
> Thank you! Mike

Hi Mike,
Also posted a request asking for exactly this. It wld be So useful to have the DepthMap and/or Normal img info as it's so easy to get confused by very similar imgs
